res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,695 @@
+    for (var i = 0x91; i <= 0x94; i++) {
+        midi.sendShortMsg(i, 0x26 - mode, OFF);
+        midi.sendShortMsg(i, 0x26 + SHIFT - mode, OFF);
+    }
+    engine.beginTimer(150, "ReloopBeatmix24.FxModeLedFlash(1, " + mode + ")",
+        true);
+};
+
+// This function activate Fx Unit 1 or 2 for the selected Channel.
+// It is mapped to SHIFT + PITCHBEND+/- (FX1 and FX2)
+ReloopBeatmix24.ActivateFx = function(channel, control, value, status, group) {
+    // Calculate Fx num based on midi control (0x66 for Fx1 and 0x67 for Fx2)
+    var FxNum = control - 0x65;
+    if (value === DOWN) {
+        if (FxModeTimers[group]) {
+            engine.stopTimer(FxModeTimers[group]);
+            delete FxModeTimers[group];
+        }
+        FxModeLongPressed[group] = false;
+        FxModeTimers[group] = engine.beginTimer(1000,
+            "ReloopBeatmix24.FxModeCallback(\"" + group + "\", " + FxNum + ")",
+            true);
+    } else { // UP
+        if (FxModeLongPressed[group]) { // long press
+            // Nothing to do, this has already been done in callback function
+            FxModeLongPressed[group] = false;
+        } else { // short press
+            // stop & delete timer
+            engine.stopTimer(FxModeTimers[group]);
+            delete FxModeTimers[group];
+            script.toggleControl("[EffectRack1_EffectUnit" + FxNum + "]",
+                "group_" + group + "_enable");
+        }
+    }
+};
+
+ReloopBeatmix24.FxKnobTurn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        var parameter = control;
+        engine.setParameter(group, "parameter" + parameter.toString(),
+            script.absoluteLin(value, 0, 1));
+    }
+    // Nothing in multi-effect mode
+};
+
+ReloopBeatmix24.ShiftFxKnobTurn = function(channel, control, value, status,
+    group) {
+    if (FxMode == 1) {
+        var parameter = 3 + control - SHIFT;
+        engine.setParameter(group, "parameter" + parameter.toString(),
+            script.absoluteLin(value, 0, 1));
+    } else {
+        var effectUnit = parseInt(group.substr(23, 1), 10);
+        var Effect = control - SHIFT;
+        var storeIndex = "FX" + Effect.toString() + "U" + effectUnit.toString();
+        if (storeIndex in previousValue) {
+            if (value - previousValue[storeIndex] > 5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect" + Effect + "]", "next_effect", 1);
+                previousValue[storeIndex] = value;
+            } else if (value - previousValue[storeIndex] < -5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect" + Effect + "]", "prev_effect", 1);
+                previousValue[storeIndex] = value;
+            }
+        } else {
+            previousValue[storeIndex] = value;
+        }
+    }
+};
+
+ReloopBeatmix24.FxOff = function(channel, control, value, status, group) {
```

Comment:
The three effects knobs have strange behaviour. When your turn them, you get MIDI Knob messages (0xB), but when they are at zero (completely on the left), they send a midi button message (0x9) with value 0x00.
When you turn them to the right (leaving the most left position), they send a midi button message (0x9) along with the first knob value (0xB).